proto/viam/app/v1/app.proto

```diff
@@ -830,6 +832,57 @@ message CheckPermissionsResponse {
   repeated AuthorizedPermissions authorized_permissions = 1;
 }
 
+message ModuleVersion {
```

Comment:
made a new message (that is essentially a duplicate of `VersionHistory`) to make this a little more flexible. 
is it possible to deprecate `VersionHistory` when modules are deprecated? if not i will reuse `VersionHistory`

Reply:
I think this is a good move long term. This is named better than `VersionHistory`.

What do you think about the name `ModuleVersionMetadata` to clearly indicate that it belongs as a subtype of module metadata?

Reply:
i like `ModuleVersion` because it is clear that it represents a version of a module (whereas it's not clear what "VersionMetadata" is). I think we can rely on docs / product communication for a user to understand that module versions are within a registry's module metadata

Reply:
Naming makes sense to me, the version fields here are specific to modules.